### PR TITLE
examples: add Confluence data retrieval example Data Job

### PR DESCRIPTION
**Pull request by yonitoo**
What: 
A data job that retrieves documents from Confluence using [LangChain's ConfluenceLoader](https://python.langchain.com/docs/integrations/document_loaders/confluence).
The user sets Confluence space key to fetch data from Confluence space. 
However, if they want to fetch a single document by ID, they could do so by setting the document ID.
The Data Job consists of 2 steps: 
 - 10_fetch_confluence_document -> fetches data from a single Confluence document);
 - 20_fetch_confluence_space -> fetches data from a Confluence space.

Why: 
As part of the Vector Database ingestion initiative, we want to build the type of RAG data population pipeline a big organisation would build.

Closes: #2990 

Signed-off-by: Yoan Salambashev <yoan.salambashev@broadcom.com>

### CODE REVIEW

File: examples/confluence-data-retrieval-example/20_fetch_confluence_space.py [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+        log.error(f"Error fetching documents from Confluence: {e}")
+        return None
+
+
+def run(job_input: IJobInput):
+    log.info(f"Starting job step {__name__}")
+
+    confluence_url = os.environ.get("VDK_CONFLUENCE_URL")
+    token = os.environ.get("VDK_CONFLUENCE_TOKEN")
+    space_key = os.environ.get("VDK_CONFLUENCE_SPACE_KEY")
```
Comment (duyguHsnHsn) [inline examples/confluence-data-retrieval-example/20_fetch_confluence_space.py:31]: What is this space key for?

File: examples/confluence-data-retrieval-example/20_fetch_confluence_space.py [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+log = logging.getLogger(__name__)
+
+
+def fetch_confluence_space(url, token, space_key):
+    try:
+        # For more info regarding the LangChain ConfluenceLoader:
+        # https://python.langchain.com/docs/integrations/document_loaders/confluence
+        loader = ConfluenceLoader(url=url, token=token)
+        documents = loader.load(
+            space_key=space_key, include_attachments=True, limit=20, max_pages=20
```
Comment (duyguHsnHsn) [inline examples/confluence-data-retrieval-example/20_fetch_confluence_space.py:18]: here why do we have this limit?

**Review by duyguHsnHsn — APPROVED** [review]
It looks good to me. I left two questions

File: examples/confluence-data-retrieval-example/20_fetch_confluence_space.py (hunk 2 above)
Comment (yonitoo) [reply examples/confluence-data-retrieval-example/20_fetch_confluence_space.py:18]: It's an example, I didn't want to make it too big but also wanted to show that we could limit it to a number we want.

File: examples/confluence-data-retrieval-example/20_fetch_confluence_space.py (hunk 1 above)
Comment (yonitoo) [reply examples/confluence-data-retrieval-example/20_fetch_confluence_space.py:31]: The Confluence spaces are a collection of related pages. Most often, for the different projects there are separate spaces. We would use the space key to fetch data from the specified space.

**Review by gageorgiev — APPROVED** [review]

**Pull request merged**